advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/InspectCommand.java

```diff
@@ -0,0 +1,166 @@
+package com.hm.achievement.command.executable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.commons.lang3.text.StrBuilder;
+import org.bukkit.ChatColor;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.command.CommandSender;
+
+import com.hm.achievement.AdvancedAchievements;
+import com.hm.achievement.command.pagination.SupplierCommandPagination;
+import com.hm.achievement.db.AbstractDatabaseManager;
+import com.hm.achievement.db.data.AwardedDBAchievement;
+import com.hm.achievement.lang.LangHelper;
+import com.hm.achievement.lang.command.CmdLang;
+import com.hm.mcshared.file.CommentedYamlConfiguration;
+
+/**
+ * Class in charge of displaying recipients of an achievement (/aach inspect).
+ *
+ * @author Rsl1122
+ */
+@Singleton
+@CommandSpec(name = "inspect", permission = "inspect", minArgs = 1, maxArgs = 5)
+public class InspectCommand extends AbstractCommand {
+
+	private static final int CACHE_EXPIRATION_DELAY = 60000;
+	private static final int PER_PAGE = 16;
+
+	private final AdvancedAchievements advancedAchievements;
+	private final AbstractDatabaseManager databaseManager;
+
+	private Map<String, Long> lastCached;
+	private Map<String, SupplierCommandPagination> cachedPaginations;
+
+	private final Map<String, String> achievementsAndDisplayNames;
+	private Map<String, String> achievementDisplayNamesAndDatabaseNames;
+
+	@Inject
+	public InspectCommand(@Named("main") CommentedYamlConfiguration mainConfig,
+			@Named("lang") CommentedYamlConfiguration langConfig, StringBuilder pluginHeader,
+			AdvancedAchievements advancedAchievements,
+			AbstractDatabaseManager databaseManager, Map<String, String> achievementsAndDisplayNames) {
+		super(mainConfig, langConfig, pluginHeader);
+		this.advancedAchievements = advancedAchievements;
+		this.databaseManager = databaseManager;
+		this.achievementsAndDisplayNames = achievementsAndDisplayNames;
+
+		this.lastCached = new HashMap<>();
+		this.cachedPaginations = new HashMap<>();
+	}
+
+	private Map<String, String> reverseMap(Map<String, String> map) {
+		Map<String, String> reversed = new HashMap<>();
+		for (Map.Entry<String, String> entry : map.entrySet()) {
+			reversed.put(entry.getValue().toLowerCase(), entry.getKey());
```

Comment:
Old versions of the plugins only worked with `Name`, so `DisplayName` might not be present in the configuration, in which case the key for the reversed map should be the `Name` key. Something like this:
```
String reversedKey = entry.getValue().isEmpty() ? entry.getKey() : entry.getValue();
reversed.put(reversedKey.toLowerCase(), entry.getKey());
```